frontend/src/book-read/components/shared/review-mutation.jsx

```diff
@@ -0,0 +1,138 @@
+import {
+  //   Box,
+  Paper,
+  Rating,
+  TextField,
+  Stack,
+  Button,
+  Typography,
+  Container,
+  Alert,
+} from "@mui/material";
+import { useState, useEffect } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+
+const initialFormState = {
+  score: 0,
+  title: "",
+  text: "",
+};
+
+export default function ReviewMutationComponent({
+  onSubmit,
+  //   onDelete,
+  //   headerTitle,
+  mutationTitle,
+  error,
+  review,
+  //   isLoading,
+  isSuccess,
+  isError,
+}) {
+  const [formState, setFormState] = useState(initialFormState);
+  const { bookId } = useParams();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (isSuccess) {
+      navigate(`/read/books/${bookId}`);
+    }
+  }, [isSuccess, navigate]);
+
+  if (review) {
+    const { score, title, text } = review;
+    useEffect(() => {
+      setFormState({
+        score,
+        title,
+        text,
+      });
+    }, [score, title, text]);
+  }
+
+  function handleRatingChange(event, value) {
+    setFormState({
+      ...formState,
+      score: value,
+    });
+  }
+
+  function handleTitleChange(event) {
+    setFormState({
+      ...formState,
+      title: event.target.value,
+    });
+  }
+
+  function handleTextChange(event) {
+    setFormState({
+      ...formState,
+      text: event.target.value,
+    });
+  }
+
+  function handleSubmit() {
+    onSubmit({ ...formState, bookId });
+  }
+
+  return (
+    <Container sx={{ width: "55%" }}>
+      <Paper
+        sx={{
+          paddingLeft: 4,
+          paddingRight: 4,
+          paddingTop: 2,
+          paddingBottom: 3,
+        }}
+      >
+        <Stack direction={"column"} spacing={3} alignItems={"center"}>
+          <Typography component="legend" sx={{ fontWeight: "bold" }}>
+            Score
+          </Typography>
+          <Rating
+            name={"score"}
+            size="large"
+            value={formState.score}
+            onChange={handleRatingChange}
+          />
+          <TextField
+            required
+            name={"title"}
+            id="outlined-basic"
+            label="Title"
+            variant="outlined"
+            size="large"
+            onChange={handleTitleChange}
+            value={formState.title}
+            fullWidth
+          />
+          <TextField
```

Comment:
This field should support writing and displaying Markdown text. It is easy to provide. 
Check this component for example:
https://github.com/TellFruit/wheelingful/blob/master/frontend/src/book-publish/components/shared/chapter-mutation.component.jsx

Reply:
Implemented